runtime/restart/restart_test.go

```diff
@@ -0,0 +1,198 @@
+		},
+		{
+			policy: "always",
+			want:   &Policy{name: "always"},
+		},
+		{
+			policy: "always:3",
+			want:   &Policy{name: "always"},
+		},
+		{
+			policy: "on-failure",
+			want:   &Policy{name: "on-failure"},
+		},
+		{
+			policy: "on-failure:10",
+			want: &Policy{
+				name:              "on-failure",
+				maximumRetryCount: 10,
+			},
+		},
+		{
+			policy: "unless-stopped",
+			want: &Policy{
+				name: "unless-stopped",
+			},
+		},
+	}
+
+	for _, testCase := range tests {
+		result, _ := NewPolicy(testCase.policy)
+		assert.Equal(t, testCase.want, result)
+	}
+}
+
+func TestRestartPolicyToString(t *testing.T) {
+	tests := []struct {
+		policy string
+		want   string
+	}{
+		{
+			policy: "",
+			want:   "always",
+		},
+		{
+			policy: "always",
+			want:   "always",
+		},
+		{
+			policy: "always:3",
+			want:   "always",
+		},
+		{
+			policy: "on-failure",
+			want:   "on-failure",
+		},
+		{
+			policy: "on-failure:10",
+			want:   "on-failure:10",
+		},
+		{
+			policy: "unless-stopped",
+			want:   "unless-stopped",
+		},
+	}
+
+	for _, testCase := range tests {
+		policy, err := NewPolicy(testCase.policy)
+		if err != nil {
+			t.Fatal(err)
+		}
+		result := policy.String()
+		assert.Equal(t, testCase.want, result)
+	}
+}
+
+func TestRestartPolicyReconcile(t *testing.T) {
+	tests := []struct {
+		policy       Policy
+		status       containerd.Status
+		restartCount int
+		want         bool
+	}{
+		{
+			policy: Policy{
+				name: "always",
+			},
+			status: containerd.Status{
+				Status: containerd.Stopped,
+			},
+			restartCount: 0,
+			want:         true,
+		},
+		{
+			policy: Policy{
+				name: "always",
+			},
+			status: containerd.Status{
+				Status: containerd.Unknown,
+			},
+			restartCount: 10,
+			want:         true,
+		},
+		{
+			policy: Policy{
+				name:              "on-failure",
+				maximumRetryCount: 10,
+			},
+			status: containerd.Status{
+				Status: containerd.Stopped,
+			},
+			restartCount: 1,
+			want:         false,
+		},
+		{
+			policy: Policy{
+				name:              "on-failure",
+				maximumRetryCount: 10,
+			},
+			status: containerd.Status{
+				Status:     containerd.Unknown,
+				ExitStatus: 1,
+			},
+			restartCount: 1,
+			want:         true,
+		},
+		{
+			policy: Policy{
+				name:              "on-failure",
+				maximumRetryCount: 3,
+			},
+			status: containerd.Status{
+				Status:     containerd.Unknown,
+				ExitStatus: 1,
+			},
+			restartCount: 3,
+			want:         false,
+		},
+		{
+			policy: Policy{
+				name: "unless-stopped",
+			},
+			status: containerd.Status{
+				Status: containerd.Unknown,
+			},
+			want: true,
+		},
+		{
+			policy: Policy{
+				name: "unless-stopped",
+			},
+			status: containerd.Status{
+				Status: containerd.Stopped,
+			},
+			want: false,
+		},
+	}
+	for _, testCase := range tests {
+		result := testCase.policy.Reconcile(testCase.status, testCase.restartCount)
+		assert.Equal(t, testCase.want, result)
+	}
+}
```

Comment:
Can we have integration tests too?
https://github.com/containerd/containerd/blob/d394e00c7eb9ed22b842f0d75c20e9353dccf488/integration/client/restart_monitor_test.go#L119